query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala

```diff
@@ -57,7 +57,7 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
         fc.parameters.head
       case fc: FunctionCall =>
         // recurse in case we have a function on top of an aggregation
-        fc.copy(parameters = fc.parameters.map(p => removeAggregates(p)))
+        fc.copy(parameters = fc.parameters.map(p => removeAggregates(p)), window = fc.window)
```

Comment:
Shouldn't need this change (or the other places where `copy` is modified just to add the `window` parameter).  Was it complaining about them?  I can't think of reason it would...

Reply:
Yes [info] Compiling 2 Scala sources to /Users/chitang/dev/query-coordinator/query-coordinator/target/scala-2.12/classes ...
[error] /Users/chitang/dev/query-coordinator/query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala:60:16: not enough arguments for method copy: (function: com.socrata.soql.functions.MonomorphicFunction[T], parameters: Seq[com.socrata.soql.typed.CoreExpr[C,T]], window: Option[com.socrata.soql.typed.WindowFunctionInfo[C,T]])com.socrata.soql.typed.FunctionCall[C,T].
[error] Unspecified value parameter window.
[error]         fc.copy(parameters = fc.parameters.map(p => removeAggregates(p)))
[error]                ^
[error] one error found
[error] (queryCoordinator / Compile / compileIncremental) Compilation failed

Reply:
Oh!  It has a manually-written copy method.  This, over in CoreExpr in soql-reference:
```scala
  def copy[C, T](function: MonomorphicFunction[T] = function,
                 parameters: Seq[CoreExpr[C, T]] = parameters,
                 window: Option[WindowFunctionInfo[C, T]]): FunctionCall[C, T] =
    FunctionCall(function, parameters, window)(position, functionNamePosition)
```
needs to be
```scala
  def copy[C, T](function: MonomorphicFunction[T] = function,
                 parameters: Seq[CoreExpr[C, T]] = parameters,
                 window: Option[WindowFunctionInfo[C, T]] = window): FunctionCall[C, T] =
    FunctionCall(function, parameters, window)(position, functionNamePosition)
```